presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java

```diff
@@ -264,6 +267,7 @@ private static VariableStatsEstimate addColumnStats(
                 .setStatisticsRange(sum)
                 .setAverageRowSize(newAverageRowSize)
                 .setNullsFraction(newNullsFraction)
+                .setHistogram(DisjointRangeDomainHistogram.addDisjunction(leftStats.getHistogram(), rightRange))
```

Comment:
The `RangeAdditionStrategy` _almost_ always expands the entire histogram range - The callees all reference methods from `StatisticRange#addSstatsAnd<X>DistinctValues` (or so I thought). It's just the distinct values that change with the strategy, so it wouldn't have an effect on the histogram's output.

See https://github.com/prestodb/presto/blob/61794f05632a8bf84a7f3b2b97916e127d403713/presto-main/src/main/java/com/facebook/presto/cost/StatisticRange.java#L152-L173

Now that you mentioned this though, I did see `addStatsAndIntersect` which doesn't actually always expand the range. So I will need to handle this case. Though it's only called when an `IntersectNode` is in the plan, which I don't believe any of the standard benchmarks generate.

The latest commit fixes this by introducing an enum wrapper around the function instead. We now check for `INTERSECT` to do the conjunction, otherwise use disjunction.

Reply:
See 

- https://github.com/ZacBlanco/prestodb/blob/7c44a1ea7fd3dc0a296e719d9b559c02bf65da09/presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java#L152-L169
- https://github.com/ZacBlanco/prestodb/blob/7c44a1ea7fd3dc0a296e719d9b559c02bf65da09/presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java#L287-L289

Reply:
Should we add tests `TestPlanNodeStatsEstimateMath` to prove correctness ?

Reply:
done